### ISSUE TITLE
Consider using Dagger to inject dependencies

### ISSUE THREAD

**Issue by cpfeiffer** [Freeyourgadget/Gadgetbridge#207]
This would allow us to get rid of the static methods in the GB class, and move the many singletons out of GBApplication.

At the same time, this would make our classes easier testable, because we could easly substitute dependencies with mocks, which is not possible with statics.

https://square.github.io/dagger/

**Comment by bytehala** [comment]
I'm interested to work on this.

**Comment by cpfeiffer** [comment]
Cool, +1!

As this will be a change that affects pretty much every class in gadgetbridge, I'd like to delay this until after we have the db-refactoring branch done and merged (port to greendao and hugely extend the db schema).

I plan to work on this next week, but if someone beats me, I'll be happy, to :-)

**Comment by bytehala** [comment]
I agree, yeah I can take a look around db-refactoring while I wait for it to be merged.

After that, we can take a staged approach to implementing Dagger 2, i.e. lay down the groundwork for dependency injection and slowly transform the classes that need it. Looking forward :-)

**Comment by cpfeiffer** [comment]
That's a very good idea!
